### ISSUE TITLE
Failed to fetch log file from worker.

### ISSUE THREAD

**Issue by smedaventera** [puckel/docker-airflow#497]
Issue: Getting below error.

`*** Log file does not exist: /usr/local/airflow/logs/LOCATION/1.log
*** Fetching from: http://CONTAINER_ID:PORT_NUMBER/log/LOCATION/1.log
*** Failed to fetch log file from worker. HTTPConnectionPool(host='CONTAINER_ID', port=PORT_NUMBER): Max retries exceeded with url: /log/LOCATION/1.log (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0xCONTAINER_ID>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution'))`

Environment details:
1. Using in AWS ECS cluster with EC2 instances.
2. OS - CentOS Linux release 7.7.1908 (Core)
3. Docker Version - Tried using both **_18.06.1.ce-3.el7_** and **_18.09.9-3.el7_**. Still getting same issue.

Note: Tried 1.10.7, 1.10.6 and 1.10.3. Facing same result.

Please help.

**Comment by anteverse** [comment]
@smedaventera Are the webserver and worker running on the same instance?
What kind of network mode are you using on ECS?

Failure in name resolution means that the webserver doesn't know where to find the worker, typically this occurs when either they're not running on the same instance or not part of the same network.

**Comment by smedaventera** [comment]
@anteverse Sorry for the delayed response.  I'm running ECS with EC2.  I only have one instance running under the ECS cluster.  And all Webserver, Worker, Flower and Scheduler are running in the same instance.  I'm not sure if that will be a problem.  Tried all possibilities that I can think of but no luck.

**Comment by anteverse** [comment]
@smedaventera Within the **containerDefinitions**, for the webserver, have you tried settings things like these:
```json
"networkMode": "bridge",
"links": [
	"airflow_worker"
],
```
?

**airflow_worker** is the container name given to the worker, defined through the key **container_name** in the **containerDefinitions** settings.

Feel free to check the AWS doc for networkModes: https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_definition_parameters.html#network_mode
I would recommend **awsvpc** even if the example shows **bridge**.

**Comment by smedaventera** [comment]
I tried all awsvpc, bridge and host in docker-compose in the Webserver.  Also, added links to worker, scheduler and flower.  But, none of them seem to help.

To give a little bit more history.  We were using 1.10.2 initially and everything was working fine.  When the 1.10.3 was release puckel republished the lower versions to address an issue which broke our docker-compose.  This was in early 2019.  After that we were using the last built image out of 1.10.2 all this while and it still is working fine in other environments.  Now we want to try the latest version and for this all that I did is to switch 1.10.2 to 1.10.7 or 1.10.6 in the Dockerfile.  Everything else remains the same.  No other configuration (ECS, EC2, Docker) has changed.  Not sure whats causing this.

**Comment by smedaventera** [comment]
Sorry for the delayed response.  I figured out the issue and now it is all working fine.  None of these changes are required.  All I had to do is - "AIRFLOW__CORE__REMOTE_LOGGING=True" set this flag in the environment variables.  We were using 1.10.2 before and now are using the 1.10.9.  For some weird reason without this flag, it was working in 1.10.2.

**Comment by amith1893** [comment]
@smedaventera  Shouldn't you provide AIRFLOW_CORE__REMOTE_BASE_LOG_FOLDER as well ?
Also, did you provide any Azure Blob Storage, Elasticsearch or Google Cloud to store the logs ?

**Comment by Inadel** [comment]
I don't know why but i my Case i Got this error when running Airflow as CeleryExecutor on Ubuntu EC2 machine (AWS)

```
*** Log file does not exist: /home/ubuntu/airflow/logs/dag_1_re.crawler/test_db_connection/2021-04-03T03:00:00+00:00/1.log
*** Fetching from: http://:8793/log/dag_1_re.crawler/test_db_connection/2021-04-03T03:00:00+00:00/1.log
*** Failed to fetch log file from worker. Invalid URL 'http://:8793/log/dag_1_re.crawler/test_db_connection/2021-04-03T03:00:00+00:00/1.log': No host supplied
```

**Comment by hafid-d** [comment]
@Inadel getting the same issue with CeleryExecutor on AWS, any idea what the problem is ? thanks

**Comment by tatitati** [comment]
Same problem here

**Comment by maslick** [comment]
Using Remote logging as proposed by @smedaventera doesn't really solve the problem, during task execution you will still be unable to read the logs. Only after the task is completed Airflow pushes the logs to S3 and then Airflow webserver can display the remote logs (S3).
